src/Components/GEOSldas_GridComp/GEOSlandassim_GridComp/clsm_ensupd_upd_routines.F90

```diff
@@ -4000,6 +4053,225 @@ subroutine cat_enkf_increments(                               &
+               N_obs,            Observations,                        &
+               1,                select_tilenum,                      &
+               N_select_species, select_species(1:N_select_species),  &
+               N_selected_obs,   ind_obs )
+          
+          if (N_selected_obs == 1) then
+             
+             do n_e=1,N_ens ! for each ensemble member
+                
+                !if(logit) write(logunit,*) 'MODIS_SCF = ', Observations(ind_obs(1))%obs
+                !if(logit) write(logunit,*) 'modeled_SCF = ', asnow_ensavg(n)
+                !if(logit) write(logunit,*) 'RHS= ', Observations(ind_obs(1))%obs*model_threshold
+                
+                if(asnow_ensavg(n) < Observations(ind_obs(1))%obs*model_threshold) then
+
+                   ! *** Add snow ***
+                   
+                   do i=1,N_snow
+                      
+                      ! a) Snow water equivalent (add equally the snow to all three layers)
+                      
+                      cat_progn_incr(n, n_e)%wesn(i) = &
+                           ( Observations(ind_obs(1))%obs - &
+                           asnow_ensavg(n)/model_threshold ) * (wesn_added/float(N_snow))
+                      
+                      !if(logit) write(logunit,*) 'cat_progn_incr(n_n_e)%wesn = ', cat_progn_incr(n, n_e)%wesn(i)
+                      
+                      ! b) snow depth 
+
+                      !if(logit) write(logunit,*) 'cat_progn(n,n_e)%sndz(i)', cat_progn(n,n_e)%sndz(i)
+
+                      if(cat_progn(n,n_e)%sndz(i) > eps) then
+                         
+                         ! Determine the fractional snow coverage
+                         wesn_sum_old(n,n_e) = sum(cat_progn(n,n_e)%wesn)
+                         areasc_old(  n,n_e) = min(wesn_sum_old(n,n_e)/wemin,1.)
+                         
+                         ! if(logit) write(logunit,*) 'wesn_sum_old(n,n_e)', wesn_sum_old(n,n_e)
+                         ! if(logit) write(logunit,*) 'areasc_old(n,n_e)', areasc_old(n,n_e)
+                         
+                         ! Estimate the density of the ld layers of snow
+                         dens_layer = max(cat_progn(n,n_e)%wesn(i)/ &
+                              (cat_progn(n,n_e)%sndz(i)*areasc_old(n,n_e)),RHOFS)
+                         
+                         !if(logit) write(logunit,*) 'dens_layer', dens_layer
+                         
+                         cat_progn_incr(n,n_e)%sndz(i)=&
+                              cat_progn_incr(n,n_e)%wesn(i)/dens_layer
+                         
+                         !if(logit) write(logunit,*) 'cat_progn_incr(n,n_e)%sndz:', cat_progn_incr(n,n_e)%sndz(i)
+                         
+                      else
+                         dens_layer = RHOFS
+                         cat_progn_incr(n,n_e)%sndz(i) = cat_progn_incr(n,n_e)%wesn(i)/dens_layer
```

Comment:
The equations that determine `cat_progn_incr(n,n_e)` here look suspicious.  Specifically, in the HISTORY output, "snow depth" is the depth of the snow within the snow covered fraction of the grid cell only, whereas "snow mass" is relative to the entire grid cell area (incl the snow-free and snow-covered portions).  See [MERRA-2 FAQ no. 2](https://gmao.gsfc.nasa.gov/reanalysis/MERRA-2/FAQ/#Q2).  I can't guarantee but I'm pretty sure that the same is true for the tile-space variables (%sndz, %wesn) here.  If so, then I would have expected "asnow" (forecast or analysis?) in the equations for cat_progn_incr.  Something to look at in more details later.  It's very unlikely that this has anything to do with Jongmin's run failure.